rr_viz/src/ross/mission_dashboard/path_navigation_tools.py

```diff
@@ -0,0 +1,151 @@
+        self.status_text_label = QLabel('Status:')
+        self.status_text_label.setFont(QFont('Ubuntu', 10, QFont.Bold))
+        self.status_label = QLabel('')
+        self.h_layout.addWidget(self.status_text_label, 3)
+        self.h_layout.addWidget(self.status_label, 7)
+        
+        self.v_layout.addLayout(self.h_layout)
+
+        # Buttons
+        self.grid_layout = QGridLayout()
+
+        self.start_recording_button = QPushButton('Start Recording')
+        self.start_recording_button.pressed.connect(self.start_recording)
+
+        self.stop_recording_button = QPushButton('Stop Recording')
+        self.stop_recording_button.pressed.connect(self.stop_recording)
+
+        self.enable_recording_buttons(False)
+        self.set_enable_record.connect(self.enable_recording_buttons)
+
+        self.follow_saved_path_button = QPushButton('Follow Saved Path')
+        self.follow_saved_path_button.pressed.connect(self.start_following)
+
+        self.stop_following_button = QPushButton('Stop Following')
+        self.stop_following_button.pressed.connect(self.stop_following)
+
+        self.enable_following_buttons(False)
+        self.set_enable_follow.connect(self.enable_following_buttons)
+
+        self.grid_layout.addWidget(self.start_recording_button, 0, 0)
+        self.grid_layout.addWidget(self.stop_recording_button, 0, 1)
+        self.grid_layout.addWidget(self.follow_saved_path_button, 1, 0)
+        self.grid_layout.addWidget(self.stop_following_button, 1, 1)
+
+        self.v_layout.addLayout(self.grid_layout)
+
+        self.setLayout(self.v_layout)
+        
+        self.record_state_checker = rr_qt_helper.StateCheckerTimer(
+            self.is_record_up, self.set_enable_record, Hz=1./3.)
+        self.record_state_checker.start()
+
+        self.follow_state_checker = rr_qt_helper.StateCheckerTimer(
+            self.is_record_up, self.set_enable_follow, Hz=1./3.)
+        self.follow_state_checker.start()
+
+    def start_recording(self):
+        path_name, ok = QInputDialog.getText(self, "Path file name", "Specify file name to save recorded path:")
+
+        if ok:
+            if path_name != "":
+                self.file_path = string.replace(self.file_path, "path.txt", path_name + ".txt")
+            
+            self.start_recording_action.send_goal(RecordPathGoal(file_path=self.file_path))
+
+    def stop_recording(self):
+        try:
+            trig_resp = self.stop_recording_srv.call(TriggerRequest())
+            if trig_resp.success:
+                print(trig_resp.message)
+            else:
+                print(trig_resp.message)
+                msg = "Failed to call '" + self.stop_mission_srv_name + "' service"
+                print(msg)
+        except:
+            msg = "Service '" + self.stop_mission_srv_name + "' unavailable"
+            rospy.logwarn(msg)
+
+    def start_following(self):
+        file_path_to_search = string.replace(self.file_path, "/path.txt", "")
```

Comment:
Are you replacing the path here to get the directory? If yes then there is a way to do that in os, that I think can be a bit nicer:
```
dir = os.path.dirname(self.file_path)
```

You would need to double check the above probably but I think I used something like this recently.